Libplanet/Store/Trie/MerkleTrie.Path.cs

```diff
@@ -0,0 +1,195 @@
+                {
+                    builder.Add((byte)(nibbles[i] << 4 | nibbles[i + 1]));
+                }
+
+                if (oddNibble > 0)
+                {
+                    builder.Add((byte)(nibbles[nibbles.Length - 1] << 4));
+                }
+
+                return new PathCursor(builder.MoveToImmutable(), nibbleOffset, nibbles.Length);
+            }
+
+            [Pure]
+            public byte NibbleAt(int index)
+            {
+                int nibbleOffset = NibbleOffset + index;
+                if (index < 0 || nibbleOffset >= NibbleLength)
+                {
+                    throw new ArgumentOutOfRangeException(nameof(index));
+                }
+
+                return GetNibble(Bytes[nibbleOffset / 2], nibbleOffset % 2 == 0);
+            }
+
+            [Pure]
+            public ImmutableArray<byte> GetRemainingNibbles()
+            {
+                var builder = ImmutableArray.CreateBuilder<byte>(RemainingNibbleLength);
+                for (int i = NibbleOffset; i < NibbleLength; i++)
+                {
+                    builder.Add(GetNibble(Bytes[i / 2], i % 2 == 0));
+                }
+
+                return builder.MoveToImmutable();
+            }
+
+            [Pure]
+            public PathCursor Next(int nibbleOffset) => nibbleOffset < 0
+                ? throw new ArgumentOutOfRangeException(nameof(nibbleOffset))
+                : new PathCursor(Bytes, NibbleOffset + nibbleOffset, NibbleLength);
+
+            // FIXME: We should declare a custom value type to represent nibbles...
+            [Pure]
+            public bool RemainingNibblesStartWith(in ImmutableArray<byte> nibbles)
+            {
+                if (NibbleLength < nibbles.Length)
+                {
+                    return false;
+                }
+
+                return CountCommonStartingNibbles(nibbles) >= nibbles.Length;
+            }
+
+            // FIXME: We should declare a custom value type to represent nibbles...
+            [Pure]
+            public int CountCommonStartingNibbles(in ImmutableArray<byte> nibbles)
+            {
+                for (int i = 0; i < nibbles.Length; i++)
+                {
+                    int nibbleOffset = NibbleOffset + i;
+                    int byteOffset = nibbleOffset / 2;
+                    if (byteOffset >= Bytes.Length)
+                    {
+                        return i;
+                    }
+
+                    byte nibble = GetNibble(Bytes[byteOffset], nibbleOffset % 2 == 0);
+                    if (nibble != nibbles[i])
+                    {
+                        return i;
+                    }
+                }
+
+                return nibbles.Length;
+            }
+
+            // FIXME: We should declare a custom value type to represent nibbles...
+            [Pure]
+            private static byte GetNibble(byte word, bool high) =>
```

Comment:
This may be not important so I'll approve this pull request, but I want to comment about the `word` word.
In my thought, It looks good about the `nibble` word usage.
Then, in my memory, the `word` unit can be different among multiple architectures. So how about using `octet` or `@byte`?
While writing this comment, I also thought that the `word` unit may mean 8-bit in .NET runtime. 🤔 

Originally I leave this comment to suggest but it seems like a question. 😅 How do you think about it?

Reply:
As you said, either *word* or *byte* can be shorter or longer than 8 bits in theory,  there virtually no more exist modern architectures using 7-bit byte or 16-bit word though.  Why I choose the term *word* here was just for avoiding the language keyword `byte`, but *octet` would be indeed the most appropriate here.  I'm going to rename it!

Reply:
@moreal Changed in 30cea86db8044731337caec3046179b65bb1040a.  Please take a look again!